### PR TITLE
feat: version tag edit

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
On model version page, a new version tag edit panel is add to allow user to manage tags of a model version

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/model_version/version_information.tsx [hunk 1]
```diff
@@ -13,35 +13,50 @@ import {
...
+    <EuiPanel data-test-subj="ml-versionInformationPanel" style={{ padding: 20 }}>
       <EuiFlexGroup alignItems="center">
         <EuiFlexItem>
           <EuiTitle size="s">
             <h3>Version Information</h3>
           </EuiTitle>
         </EuiFlexItem>
         <EuiFlexItem style={{ marginLeft: 'auto', flexGrow: 0 }}>
-          {readOnly ? (
+          {readOnly || isEditDisabled ? (
```
Comment (wanglam) [inline public/components/model_version/version_information.tsx:59]: The current edit button disable is a little bit strange for me. Have you check this with designer? I am prefer disable the edit button immediately after another edit button clicked. The disable logic for now need form changed, is that aligned with designer?

File: public/components/common/forms/model_tag_array_field/tag_field.tsx [hunk 2]
```diff
@@ -228,97 +240,127 @@ export const ModelTagField = ({
...
+          grow={false}
+          style={index === 0 ? { transform: 'translateY(22px)' } : undefined}
+        >
+          <EuiToolTip content={tags?.length && tags.length > 1 ? 'Remove' : 'Clear'}>
+            <EuiButtonIcon
+              display="base"
+              size="m"
+              iconType="cross"
+              aria-label={`Remove tag at row ${index + 1}`}
+              onClick={() => onRemove(index)}
```
Comment (wanglam) [inline public/components/common/forms/model_tag_array_field/tag_field.tsx:359]: The delete confirmation modal for default tags missed. Will we added it in the future PR?
image

Comment (ruanyl) [reply public/components/common/forms/model_tag_array_field/tag_field.tsx:359]: Yes, I will add it later PR

File: public/components/model_version/version_information.tsx (hunk 1 above)
Comment (ruanyl) [reply public/components/model_version/version_information.tsx:59]: I will check with the UX, I can update this in upcoming PRs in needed.

**Comment by wanglam** [comment]
@ruanyl Seems there is a lint error need to be fixed https://github.com/opensearch-project/ml-commons-dashboards/actions/runs/5010825690/jobs/8981052395?pr=182

**Review by wanglam — APPROVED** [review]

**Pull request merged**